src/Core/Geometry/Image.cpp

```diff
@@ -296,4 +303,37 @@ std::shared_ptr<Image> FlipImage(const Image &input)
     return output;
 }
 
+std::shared_ptr<Image> DilateImage(const Image &input,
+        int half_kernel_size/* = 1 */)
+{
+    auto output = std::make_shared<Image>();
+    if (input.num_of_channels_ != 1 || input.bytes_per_channel_ != 1) {
+        PrintWarning("[DilateImage] Unsupported image format.\n");
+        return output;
+    }
+    output->PrepareImage(input.width_, input.height_, 1, 1);
+
+#ifdef _OPENMP
+#pragma omp parallel for schedule(static)
+#endif
+    for (int y = 0; y < input.height_; y++) {
+        for (int x = 0; x < input.width_; x++) {
+            for (int yy = -half_kernel_size; yy <= half_kernel_size; yy++) {
+                for (int xx = -half_kernel_size; xx <= half_kernel_size; xx++) {
+                    unsigned char* pi;
+                    if (input.TestImageBoundary(x+xx, y+yy)) {
+                        pi = PointerAt<unsigned char>(input, x+xx, y+yy);
+                        if (*pi == 255) {
+                            *PointerAt<unsigned char>(*output, x, y, 0) = 255;
+                            xx = half_kernel_size;
+                            yy = half_kernel_size;
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
I am not quite like this function. All the filtering functions in Open3D are implemented using 1D domain to save computational burden, but for this function, I had to use 4 nested for-loops. Maybe there would be better implementation for this.

Reply:
I think it is fine for now. In the long run, we should seriously consider how to use OpenCV in Open3D. Actually I think [vcpkg](https://github.com/Microsoft/vcpkg) might be our savior. All that is missing is a package management system on Windows. Thus we had to include all the source code and are afraid of linking to big packages like OpenCV. But if Windows have a package management system, lots of our problems will be solved. We should invest time in this. @syncle @takanokage